### ISSUE TITLE
Declaring an extern function or task in a class is marked as error

### ISSUE THREAD

**Issue by amey-kulkarni** [eirikpre/VSCode-SystemVerilog#205]
When declaring an extern function or task, the function or task keyword is highlighted as an error because there is no endfunction or endtask keyword associated with it.
For example:

```
class MyClass:
    extern function void MyFunction();
endclass : MyClass

function void MyClass::MyFunction();
    foo;
endfunction : MyFunction
```

In the above example, the extern function declaration is highlighted as an error.

**Comment by joecrop** [comment]
@amey-kulkarni, you are referring to the word `function` being assigned red text, right? The ANTLR parser as well as other compilers don't seem to have an issue with this syntax, from what I can tell. As a matter of fact, the syntax highliter itself is fine with this too.

The issue is coming from [language-configuration.json](https://github.com/eirikpre/VSCode-SystemVerilog/blob/3528b778bd60eedf7c54be5b7fce40c7adb44ce5/language-configuration.json#L24C9-L24C37). This list tells vscode about other symbols that should also be treated as begin/end brackets. Putting function/endfunction on this list allows vscode to collapse functions, jump the cursor, etc. It also makes it hand to remind users that they may have forgotten an endfunction.  It is also what makes the word `function` purple instead of the default blue. The downside is that we can't make this bracket mode syntax-aware/smart.

Long story short, it seems to me that the convince/power of listing function/endfunction as a 'bracket' outweighs the red text in cases where endfunction is not needed.
